### PR TITLE
TPF-1657 -> master | Update C2FO ui-guide for 2017, consume it in new supplier frontend

### PR DESCRIPTION
**Pull request by cmgurba**
[JIRA Link](https://honeycomb.jira.com/browse/TPF-1657)  
## Description    
We need to share styles across c2fo proper and the new lending supplier app to keep a consistent theming/branding approach.   There has been work towards accomplishing this in https://github.com/c2fo/ui-guide, but it has not been under maintenance as we went in favor of including the ui guide directly in the main c2fo repository.  This, however, means it cannot be consumed as an npm package by multiple apps which is what we want.  We want to keep it to basic shared css, agnostic of javascript or style frameworks.

Acceptance Criteria:
* update existing repo or create new one (and by request possibly give it a more specific name than ui-guide) to achieve the below:
* port over styles within the c2fo repo (https://github.com/C2FO/c2fo/tree/master/c2fo-frontend/assets/styles/ui-guide) to the separate repo, using build tools to:
** Export separate variables.less and variables.scss for consuming applications to use depending on their preprecessor of choice
** Export a "styles.min.css" containing the rest of the common functionality
* Host the style guide website using github pages that is currently contained at https://app.c2fo.com/styleguide with any updates necessary after the changes.

## Test Steps    
Put your test steps here....

### CODE REVIEW

File: src/less/alerts.less [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+/* ------------------------------------------------------------------------- *\
+    #BaseAlert
+
+
+    All alerts will be described with a role atribute value of "alert" or "alertdialog"
+    if the alert contains actions (see #AlertActions).
+
+    1. Clears the floats
+    2. Align border-radius, margins, and ul, p and a tag styles with bootstrap basics.
```
Comment (cmgurba) [inline src/less/alerts.less]: I was hesitant of this as we are trying to be framework agnostic yet our current alerts are very tied in with bootstrap.  I guess these can be pulled out if necessary.

File: Gruntfile.js [hunk 2]
```diff
@@ -19,10 +19,23 @@ module.exports = function(grunt) {
...
         },
         files: {
-          'docs/assets/css/app.css': 'src/less/index.less'
+          'docs/assets/css/app.css': [
+            'src/less/index.less',
+            'node_modules/grid-system/src/index.less'
+          ]
+        }
+      },
+      dist: {
```
Comment (cmgurba) [inline Gruntfile.js]: I thought a separate command made sense to minify and exclude sourcemap, but maybe we should just rely on the end application for that...

**Comment by cmgurba** [comment]
The idea is that c2fo-ui-guide.min.css contains all of the .less content in src/ , which will be the actual consumed content.  Maybe I should structure it to hide the less better.

**Comment by garrypolley** [comment]
>  Maybe I should document that src is not meant to be consumed

I am in agreement.

**Comment by cmgurba** [comment]
I will also be testing consuming this package today as i continue to scaffold our new frontend supplier app, so I update doc and tweak things accordingly

**Pull request merged**